source/utils/LogSinkFile.cpp

```diff
@@ -0,0 +1,60 @@
+/*
+*  Copyright (C) 2011-2015  OpenDungeons Team
+*
+*  This program is free software: you can redistribute it and/or modify
+*  it under the terms of the GNU General Public License as published by
+*  the Free Software Foundation, either version 3 of the License, or
+*  (at your option) any later version.
+*
+*  This program is distributed in the hope that it will be useful,
+*  but WITHOUT ANY WARRANTY; without even the implied warranty of
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+*  GNU General Public License for more details.
+*
+*  You should have received a copy of the GNU General Public License
+*  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "utils/LogSinkFile.h"
+
+#include "utils/Helper.h"
+
+LogSinkFile::LogSinkFile(const std::string& filepath)
+    : mFilePath(filepath)
+{
+    mFile.open(mFilePath.c_str(), std::ios::out | std::ios::trunc);
+    if (mFile.is_open())
+    {
+        mFile.close();
```

Comment:
Are you sure it is a good idea to close the file each time you write ? I'm not an expert but when I've done the file logging manager, I've opened the file in the constructor and closed in the destructor. And that because that's how it is done in Ogre. And I would think that if they do like that for such a feature, that's probably the best way to do it.
See:
https://github.com/ehsan/ogre/blob/master/OgreMain/src/OgreLog.cpp#L51